<launch>

  <arg name="uav_name" default="$(optenv UAV_NAME uav)"/>
  <arg name="gui" default="true"/>
  <arg name="debug" default="false"/>

  <group ns="$(arg uav_name)">

    <node pkg="image_transport" type="republish" name="republish_mobius" output="screen" args="compressed in:=mobius/image_raw raw out:=mobius2/image_raw" />

    <!--node name="mbzirc_detector" pkg="object_detection" type="mbzirc_detector" output="screen" launch-prefix="gdb -ex run -o-args" -->
    <node name="mbzirc_detector" pkg="object_detection" type="mbzirc_detector" output="screen">

      <rosparam file="$(find object_detection)/cfg/f550.yaml" />
      <rosparam file="$(find object_detection)/cfg/$(arg uav_name).yaml" />
      
      <param name="uav_name" type="string" value="$(arg uav_name)"/>
      <param name="gui" type="bool" value="$(arg gui)"/>
      <param name="debug" type="bool" value="$(arg debug)"/>

      <!-- Subscribed topics -->
      <remap from="~image_in" to="mobius2/image_raw" />
      <remap from="~camera_info" to="mobius/camera_info" />
      <remap from="~uav_odometry" to="mbzirc_odom/new_odom" />

      <!-- Published topics -->
      <remap from="~objectRelative" to="~objectRelative" />
      <remap from="~objectPositions" to="~objectPositions" />
      <remap from="~objects" to="~objects" />
      <remap from="~object_array" to="~object_array" />
      <remap from="~processedimage" to="~processedimage" />
      <remap from="~objectImages" to="~objectImages" />
      <remap from="~histogram" to="~histogram" />

    </node>

  </group>

</launch>
